diff --git a/src/serialmodule.c b/src/serialmodule.c
index f7f7b24..63286b0 100644
--- a/src/serialmodule.c
+++ b/src/serialmodule.c
@@ -13,7 +13,7 @@
 *		The (only) main header file to export 5 APIs: [spsr_module_init, spsr_module_finish, spsr_inst_open, spsr_inst_close, spsr_inst_write].
 */
 /*+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+*/
-#ifndef ___SIMPLE_SERIAL_MODULE__
+
 #include "serialmodule.h"
 #include <stdio.h>
 #include <simplelog.h>
